controlpanel/api/migrations/0010_DATA_add_tools.py

```diff
@@ -0,0 +1,80 @@
+# Generated by Django 2.2.4 on 2019-08-28 10:34
+
+import json
+import os
+import secrets
+
+from django.conf import settings
+from django.db import migrations
+
+
+TOOLS = [
+    {
+        'chart_name': 'rstudio',
+        'version': '2.1.1',
+        'name': 'RStudio',
+        'description': 'RStudio: 1.2.1335+conda, R: 3.5.1, Python: 3.7.1, patch: 3',
+        'values': {
+            'authProxy.cookieSecret': '<SECRET_TOKEN>',
+            'authProxy.auth0.domain': '%(domain)s',
```

Comment:
So, to be sure I understand what's going on. The `'%(name)s'` means something like this right?:

```python
template = 'my name is %(name)s'
name = "John Doe"
rendered = template % name # rendered == "my name is John Doe"
```

So it's basically like `%s` in `printf()` but using the variable name right?

Reply:
Yeah, it's the old Python string interpolation style - `'%(domain)s' % {'domain': 'some value'}` is equivalent to `'{domain}'.format({'domain': 'some value'})`. The `<SECRET_TOKEN>` string was previously `{{secret_token}}`, but you mentioned getting an error with the migration, so I changed it so that there was no chance of a syntax problem or reading comprehension fail. I could go back to the `format` style, but it's a data migration that is intended to be run only once, so no point changing it.